cmd/vcluster/cmd/start.go

```diff
@@ -66,31 +71,45 @@ func ExecuteStart(ctx context.Context, options *options.VirtualClusterOptions) e
 	}
 
 	// get current namespace
-	currentNamespace, err := clienthelper.CurrentNamespace()
+	controlPlaneConfig, controlPlaneNamespace, controlPlaneService, workloadConfig, workloadNamespace, workloadService, err := pro.GetRemoteClient(options)
 	if err != nil {
 		return err
 	}
+	options.ServiceName = workloadService
+	err = os.Setenv("NAMESPACE", workloadNamespace)
+	if err != nil {
```

Comment:
[Reduced variable scope](https://github.com/uber-go/guide/blob/master/style.md#reduce-scope-of-variables)
```suggestion
	if err := os.Setenv("NAMESPACE", workloadNamespace); err != nil {
```